src/index.js

```diff
@@ -53,42 +52,33 @@ module.exports = function createAtom (initialState = {}, actions = {}, options =
         const history = sourceActions.concat([action])
         const dispatch = createDispatch(history)
         const set = createSet(history)
-        const actionAtom = Object.assign({}, mutableAtom, { dispatch, set })
+        const swap = createSwap(history)
+        const actionAtom = Object.assign({}, atom, { dispatch, set, swap })
         return evolve(actionAtom, action, actions)
       } else {
-        return evolve(mutableAtom, action, actions)
+        return evolve(atom, action, actions)
       }
     }
   }
 
-  function createSet (sourceActions) {
+  function createSet (sourceActions, { swap = false } = {}) {
     sourceActions = sourceActions || []
     return function set (update, options = {}) {
       let action = { payload: update }
       let prevState = state
-      state = options.replace ? action.payload : merge(state, action.payload)
+      state = swap ? action.payload : Object.assign({}, state, action.payload)
```

Comment:
Do we need to make a copy for `swap`? Could run into something like:

```js
const newState = { foo: 'bar' }
atom.swap(newState)
newState.foo = 'not-bar'
```

Reply:
I'm not too worried about that, you can already mutate state with `atom.get().foo = 'not-bar'` and similar.

Swap is meant to really swap the full reference, useful, when for example, you use Immutable.js where set() with it's Object.assign doesn't make sense to ue. Previously, this was configurable via `options.merge`, but I prefer to remove that option and have these 2 apis instead.

Swap is also useful if you're doing things like removing top level keys from the state or when you're resetting the entire state from local storage, e.g. the `tiny-atom/debug` redux devtools integration does something similar when you click on a historical state entry.